plugins/auth/auth_base.js

```diff
@@ -101,20 +101,28 @@ exports.check_user = function (next, connection, credentials, method) {
         return;
     }
 
-    const passwd_ok = function (valid, message) {
+    // valid: (true|false)
+    // opts: ({ message, code }|String)
+    function passwd_ok (valid, opts) {
+        const status_code = (typeof(opts) === 'object' && opts.code) || (valid ? 235 : 535);
+        const status_message = (typeof(opts) === 'object' ? opts.message : opts) ||
+                (valid  ? '2.7.0 Authentication successful' : '5.7.8 Authentication failed');
+
         if (valid) {
             connection.relaying = true;
             connection.results.add({name:'relay'}, {pass: plugin.name});
+            connection.results.add(plugin, {pass: method});
+
             connection.results.add({name:'auth'}, {
                 pass: plugin.name,
                 method: method,
                 user: credentials[0],
             });
-            connection.respond(235, ((message) ? message : 'Authentication successful'), function () {
+
+            connection.respond(status_code, status_message, function () {
                 connection.authheader = "(authenticated bits=0)\n";
                 connection.auth_results(`auth=pass (${method.toLowerCase()})`);
                 connection.notes.auth_user = credentials[0];
-                connection.notes.auth_passwd = credentials[1];
```

Comment:
Yeah, that's why it adds in Changes.md:
```diff
+    * Breaking: auth_base.js: don't set `connection.notes.auth_passwd`
```
Anyway, I'll make it configurable, but I think we already talked about this -- that this is a bad practice. Main danger lurks in further plugins, possibly user-written, that would somehow allow attacker to execute malicious code. Once he's able to get access to notes object, user's info is compromised.

I'd recommend switching the default to REMOVE rather than LEAVE.